python/foglamp/tasks/backup_restore/backup.py

```diff
@@ -0,0 +1,667 @@
+
+__author__ = "Stefano Simonelli"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_MODULE_NAME = "foglamp_backup_module"
+
+_MESSAGES_LIST = {
+
+    # Information messages
+    "i000001": "Execution started.",
+    "i000002": "Execution completed.",
+
+    # Warning / Error messages
+    "e000001": "cannot initialize the logger - error details |{0}|",
+    "e000002": "an error occurred during the backup operation - error details |{0}|",
+}
+""" Messages used for Information, Warning and Error notice """
+
+_logger = {}
+
+
+class Backup(object):
+    """ Backup API integration
+
+    # CURRENTLY NOT IMPLEMENTED !!
+    """
+
+    # noinspection PyUnusedLocal
+    def get_backup_list(self, limit, skip, status):
+        """  Retrieves backups information
+
+        Args:
+            limit: maximum number of backups information to retrieve
+            skip: TBD - # FIXME:
+            status: BACKUP_STATUS_UNDEFINED= retrieves the information for all the backups state
+                    or for a specific state only
+        Returns:
+        Raises:
+        """
+
+        pass
+
+
+class BackupProcess(FoglampProcess):
+    """ Backups the entire FogLAMP repository into a file in the local filesystem,
+        it executes a full warm backup
+    """
+
+    _MODULE_NAME = "foglamp_backup"
+
+    _MESSAGES_LIST = {
+
+        # Information messages
+        "i000001": "Execution started.",
+        "i000002": "Execution completed.",
+
+        # Warning / Error messages
+        "e000000": "general error",
+        "e000001": "cannot initialize the logger - error details |{0}|",
+        "e000002": "cannot retrieve the configuration from the manager, trying retrieving from file "
+                   "- error details |{0}|",
+        "e000003": "cannot retrieve the configuration from file - error details |{0}|",
+        "e000004": "cannot delete/purge backup file on file system - id |{0}| - file name |{1}| error details |{2}|",
+        "e000005": "cannot delete/purge backup information on the storage layer "
+                   "- id |{0}| - file name |{1}| error details |{2}|",
+        "e000007": "backup failed.",
+        "e000008": "cannot execute the backup, either a backup or a restore is already running - pid |{0}|",
+        "e000009": "cannot retrieve information for the backup id |{0}|",
+        "e000010": "directory used to store backups doesn't exist - dir |{0}|",
+        "e000011": "directory used to store semaphores for backup/restore synchronization doesn't exist - dir |{0}|",
+        "e000012": "cannot create the configuration cache file, neither FOGLAMP_DATA or FOGLAMP_ROOT are defined.",
+        "e000013": "cannot create the configuration cache file, provided path is not a directory - dir |{0}|",
+    }
+    """ Messages used for Information, Warning and Error notice """
+
+    _CONFIG_FILE = "configuration_cache.json"
```

Comment:
The name should indicate this cache is related to the backup process in some way. There will be other json files in the directory so we should make it obvious what they all are.

Reply:
changed to : backup_postgres_configuration_cache.json